smartapps/samsung/sleepsense-fall-asleep.src/sleepsense-fall-asleep.groovy

```diff
@@ -0,0 +1,151 @@
+    iconX3Url: "https://s3.amazonaws.com/smartapp-icons/Convenience/Cat-Convenience@2x.png")
+
+
+
+preferences {
+	page(name:"Mainpage", install:true, uninstall:true) {
+        section("When you fall asleep, do this.") {
+            input "sleepsensor", "capability.sleepSensor", title: "Sleep Sensor", multiple: false, required: true
+        }
+       section("Select lights/bulbs") {
+            input "LightLevelController", "capability.switchLevel", title: "Select light", multiple: true, required: false
+            input(name: "lightlevel", type: "enum", title: "level", options: ["off", "10%","20%","30%","40%","50%","60%","70%","80%","90%","100%"])
+        }
+	}
+}
+
+
+def installed()
+{
+	log.trace "installed()"
+   	state.sleepAction = "Sleeping"
+
+    subscribe(sleepsensor, "sleeping", sleepsensorHandler)    
+}
+
+def updated()
+{
+	log.trace "updated()"
+	state.sleepAction = "Sleeping"
+    
+	unsubscribe()
+    subscribe(sleepsensor, "sleeping", sleepsensorHandler)
+    subscribe(sleepsensor, "bedstate", sleepsensorHandler)
+  
+}
+
+def initialize() {
+	// TODO: subscribe to attributes, devices, locations, etc.
+}
+
+
+def sleepsensorHandler(evt) {
+	//log.debug "sleepsensorHandler: $evt.name"	
+    log.debug "sleepsensorHandler: $evt.value"
+    state.isOK = false;
+    
+    if((state.sleepAction == "Sleeping") && (evt.value == "sleeping"))
+    {
+    	log.debug "Sleeping actions will be excuted"
+    	state.isOK = true;
+    }
+    else
+    {
+    	return
+    }
+    
+    LightLevelConv()
+    LightLevelController()
+   
+}
+
+def LightLevelConv() {
+	if (lightlevel == "off") {
+    	state.lightlevel = 0;
+    }
+	else if (lightlevel == "10%") {
+    	state.lightlevel = 10
+    }
+    else if (lightlevel == "20%") {
+    	state.lightlevel = 20
+    }
+    else if (lightlevel == "30%") {
+    	state.lightlevel = 30
+    }
+    else if (lightlevel == "40%") {
+    	state.lightlevel = 40
+    }
+    else if (lightlevel == "50%") {
+    	state.lightlevel = 50
+    }
+    else if (lightlevel == "60%") {
+    	state.lightlevel = 60
+    }
+    else if (lightlevel == "70%") {
+    	state.lightlevel = 70
+    }
+    else if (lightlevel == "80%") {
+    	state.lightlevel = 80
+    }
+    else if (lightlevel == "90%") {
+    	state.lightlevel = 90
+    }
+    else if (lightlevel == "100%") {
+    	state.lightlevel = 100
+    }
+    
+}
+
+def LightLevelController() {
+    log.debug "LightLevelController()"
+    if(state.isOK == true)
+    {    	
+    	LightLevelController.each
+        {
+            log.debug it
+            if(state.lightlevel == 0)
+            {
+	            log.debug "light off"
+    	        it?.off()
+            }
+            else
+            {
+	            it?.setLevel(state.lightlevel)
+            }
+        }        
+	}
+	log.debug state.lightlevel
+}
+
+def changedLocationMode(evt) {
+	log.debug "changedLocationMode: $evt"
+	switches?.on()
+}
+
+def appTouch(evt) {
```

Comment:
Also not sure why this is here. This doesn't actually "run" the app, which is the purpose of touching the app.